### DISCUSSION TITLE
Variables in HTML and YML files

### DISCUSSION THREAD

**Discussion by syeda-git** [fralau/mkdocs-macros-plugin#168] [Q&A]
Hi,
In my mkdocs yml file, i have defined a variable as follows:
extra:
  prodname: XYZ

In the markdown files, when i use {{prodname}}, they render correctly to XYZ in the HTML pages. However, i have a home.html file in the theme_overides folder. There is also a nav section in mkdocs.yml file. When i use {{prodname}} in home.html and mkdocs.yml, they don't render correctly to XYZ. Instead, 

Does that mean that the variables in the  macros plugin do not work on HTML and YML files?

Thanks...
Syed

**Comment by fralau** [comment]
Yes mkdocs-macros works only on markdown pages, not on YAML files.

It's important to realize that template pages do use a jinja2 templating engine, but that's a different one (belonging to mkdocs). You can "trickle" values into it, through the [metadata of the markdown page](https://mkdocs-macros-plugin.readthedocs.io/en/latest/post_production/#html-pages).
